ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/ScenarioLoader.scala

```diff
@@ -150,36 +149,42 @@ object ScenarioLoader {
   }
 
   private def getCandidateScenarios(
-      packages: DamlPackageContainer,
+      packages: InMemoryPackageStore,
       scenarioQualName: Ref.QualifiedName
   ): List[(Ref.Identifier, Definition)] = {
-    packages.packages.flatMap {
-      case (packageId, pkg) =>
-        pkg.lookupIdentifier(scenarioQualName) match {
-          case Right(x) => List((Ref.Identifier(packageId, scenarioQualName), x))
-          case Left(_) => List()
-        }
-    }(breakOut)
+    packages
+      .listLfPackagesSync()
+      .flatMap {
+        case (packageId, _) =>
+          val pkg = packages.getLfPackageSync(packageId).get
+          pkg.lookupIdentifier(scenarioQualName) match {
+            case Right(x) => List((Ref.Identifier(packageId, scenarioQualName), x))
+            case Left(_) => List()
+          }
+      }(breakOut)
   }
 
   private def getScenarioQualifiedName(
-      packages: DamlPackageContainer,
+      packages: InMemoryPackageStore,
       scenario: String
   ): Ref.QualifiedName = {
     Ref.QualifiedName.fromString(scenario) match {
       case Left(err) =>
         logger.warn(
           "Dot-separated scenario specification is deprecated. Names are Module.Name:Inner.Name, with a colon between module name and the name of the definition. Falling back to deprecated name resolution.")
-        packages.packages.iterator
+        packages
+          .listLfPackagesSync()
+          .iterator
           .map {
-            case (_, pkg) => DeprecatedIdentifier.lookup(pkg, scenario)
+            case (pkgId, _) =>
+              DeprecatedIdentifier.lookup(packages.getLfPackageSync(pkgId).get, scenario)
```

Comment:
Likewise, `.get`. Should you discard failed packages here instead? Not sure.

Reply:
It is unlikely that `.get` would fail here - this code is more or less equivalent to `myMap.keys.map(myMap.get)`. I have replaced `.get` with `.getOrElse` and a helpful error message.